requests/exceptions.py

```diff
@@ -100,6 +100,8 @@ class StreamConsumedError(RequestException, TypeError):
 class RetryError(RequestException):
     """Custom retries logic failed"""
 
+class RedirectError(RequestException):
```

Comment:
`RedirectError` seems a little nebulous, the current occurrences are more of an `IOError`, but I'm not sure what we'd want to call a Requests version of that. Open to suggestions.